### PR TITLE
Add new module `azure_rm_account_info` to get facts for current logged in user

### PR DESCRIPTION
**Pull request by mandar242**
##### SUMMARY

Add new module `azure_rm_account_info`.
Provides output equivalent to `az account show`.

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

azure_rm_account_info
##### ADDITIONAL INFORMATION

Makes use of below to create a output containing details similar to and of similar format to `az account show`
- https://docs.microsoft.com/en-us/python/api/azure-mgmt-resource/azure.mgmt.resource.subscriptions.v2019_11_01.models.subscription?view=azure-python
- https://docs.microsoft.com/en-us/python/api/azure-graphrbac/azure.graphrbac.operations.signedinuseroperations?view=azure-python#azure-graphrbac-operations-signedinuseroperations-get

### CODE REVIEW

File: plugins/modules/azure_rm_account_info.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+
+        # Makes use of azure.mgmt.subsctiption.operations.subscriptionoperations
+        # https://docs.microsoft.com/en-us/python/api/azure-mgmt-subscription/azure.mgmt.subscription.operations.subscriptionsoperations?view=azure-python#methods
+
+        try:
+            subscription_list_response = list(self.subscription_client.subscriptions.list())
+        except CloudError as exc:
+            self.fail("Failed to list all subscriptions - {0}".format(str(exc)))
+
+        results['id'] = subscription_list_response[0].subscription_id
```
Comment (abikouo) [inline plugins/modules/azure_rm_account_info.py:176]: is this a choice to return only the first subscription ``[0]`` for the account?

File: plugins/modules/azure_rm_account_info.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        return results
+
+    def get_managed_by_tenants_list(self, object_list):
+
+        result = []
+
+        for item in object_list:
+            result.append({"tenantId": item.tenant_id})
+
+        return result
```
Comment (abikouo) [inline plugins/modules/azure_rm_account_info.py]:
```suggestion
        return [dict(tenantId=item.tenant_id) for item in object_list]
```

**Review by abikouo — CHANGES_REQUESTED** [review]
@mandar242 Thanks for this pull request.
Could you please add some integration tests for this module?

File: plugins/modules/azure_rm_account_info.py (hunk 1 above)
Comment (mandar242) [reply plugins/modules/azure_rm_account_info.py:176]: Looking at the documentation and based on my trial-errors for `az account show` here: https://docs.microsoft.com/en-us/cli/azure/account?view=azure-cli-latest#az-account-show
I felt that it returns a single subscription, so adding `[0]`th subscription.
I could be wrong here though. Thoughts?

File: tests/integration/targets/azure_rm_account_info/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+
+- name: Get facts for current logged in user
+  azure.azcollection.azure_rm_account_info:
+  register: result
+
+- assert:
+    that:
+      - result is not changed
+      - result is not failed
```
Comment (abikouo) [inline tests/integration/targets/azure_rm_account_info/tasks/main.yml:8]: It should be good to test that the result contains ``account_info`` and the all expected key from account_info, ``tenantId``, ``state``, etc

**Review by abikouo — CHANGES_REQUESTED** [review]

File: tests/integration/targets/azure_rm_account_info/tasks/main.yml (hunk 3 above)
Comment (mandar242) [reply tests/integration/targets/azure_rm_account_info/tasks/main.yml:8]: Added.

File: tests/integration/targets/azure_rm_account_info/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+      - "'account_info' in result"
+      - "'environmentName' in result.account_info"
+      - "'homeTenantId' in result.account_info"
+      - "'id' in result.account_info"
+      - "'managedByTenants  ' in result.account_info"
```
Comment (Fred-sun) [inline tests/integration/targets/azure_rm_account_info/tasks/main.yml]:
```suggestion
      - "'managedByTenants' in result.account_info"
```

**Review by Fred-sun — COMMENTED** [review]
Consider deleting lines in the module that are not useful for modifying comments. Thank you very much!

File: tests/integration/targets/azure_rm_account_info/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
+---
```
Comment (Fred-sun) [inline tests/integration/targets/azure_rm_account_info/tasks/main.yml]:
```suggestion
```

File: tests/integration/targets/azure_rm_account_info/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
```
Comment (Fred-sun) [inline tests/integration/targets/azure_rm_account_info/tasks/main.yml]:
```suggestion
```

**Comment by xuzhang3** [comment]
LGTM

**Pull request merged**